pkg/coastguard/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"sync"
+
+	"github.com/submariner-io/admiral/pkg/federate"
+	"github.com/submariner-io/submariner/pkg/signals"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+
+	"github.com/submariner-io/admiral/pkg/federate/kubefed"
+
+	"github.com/submariner-io/coastguard/pkg/controller"
+)
+
+var kubeConfig string
+var masterURL string
+
+func init() {
+	flag.StringVar(&kubeConfig, "kubeconfig", os.Getenv("KUBECONFIG"),
+		"Path to kubeconfig containing embedded authinfo.")
+	flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+
+}
+
+func main() {
+	klog.InitFlags(nil)
+	flag.Parse()
+
+	klog.V(2).Info("Starting coastguard-network-policy-sync")
+
+	// set up signals so we handle the first shutdown signal gracefully
+	stopCh := signals.SetupSignalHandler()
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+
+	coastGuardController := controller.New()
+	go func() {
+		defer wg.Done()
+		coastGuardController.Run(stopCh)
+	}()
+
+	federator := buildKubeFedFederator(stopCh)
+	err := federator.WatchClusters(coastGuardController)
```

Comment:
I'm not able to locate the implementation of this API at https://github.com/kubernetes-sigs/kubefed
Can you please point me to its path.

Reply:
It's in our new [admiral repo](https://github.com/submariner-io/admiral).